DEVELOPMENT.md

```diff
@@ -0,0 +1,536 @@
+
+```bash
+# Start frontend with local backend (default)
+python3 etc/nyrkio_frontend.py start
+
+# Start frontend with production backend
+python3 etc/nyrkio_frontend.py start --mode production
+
+# Start on custom port
+python3 etc/nyrkio_frontend.py start --port 3000
+
+# Check frontend status
+python3 etc/nyrkio_frontend.py status
+
+# Stop frontend
+python3 etc/nyrkio_frontend.py stop
+
+# Restart frontend
+python3 etc/nyrkio_frontend.py restart
+
+# Install dependencies only
+python3 etc/nyrkio_frontend.py install
+```
+
+**Deployment Modes:**
+
+- **local** (default): Uses `vite.config.test.js` - proxies API requests to `http://localhost:8001`
+- **production**: Uses `vite.config.js` - proxies API requests to `https://nyrk.io`
+
+**Features:**
+- Auto-detects and installs npm dependencies
+- Runs server in background with PID tracking
+- Configurable port (default: 5173)
+- Graceful shutdown support
+
+**Frontend Configuration Files:**
+
+- `frontend/vite.config.js` - Production configuration (proxies to https://nyrk.io)
+- `frontend/vite.config.test.js` - Local development configuration (proxies to http://localhost:8001)
+
+### Using npm Directly
+
+For foreground execution with full output:
+
+```bash
+cd frontend
+
+# Install dependencies (first time only)
+npm install
+
+# Start with local config
+npm run dev -- --config vite.config.test.js --port 5173
+
+# Start with production config
+npm run dev -- --config vite.config.js --port 5173
+```
+
+**Default URLs:**
+- Frontend: http://localhost:5173
+- API (local): http://localhost:8001
+- API (production): https://nyrk.io
+
+### Full Stack Development
+
+For a complete local development environment:
+
+```bash
+# Terminal 1: Start backend
+python3 etc/nyrkio_backend.py start
+
+# Terminal 2: Start frontend in local mode
+python3 etc/nyrkio_frontend.py start
+
+# Or run both and check status
+python3 etc/nyrkio_backend.py status
+python3 etc/nyrkio_frontend.py status
+```
+
+## Running the Backend
+
+### Using nyrkio_backend.py (Recommended)
+
+The `etc/nyrkio_backend.py` script manages the backend server process:
+
+```bash
+# Start the backend server
+python3 etc/nyrkio_backend.py start
+
+# Check if backend is running
+python3 etc/nyrkio_backend.py status
+
+# Stop the backend server
+python3 etc/nyrkio_backend.py stop
+
+# Restart the backend server
+python3 etc/nyrkio_backend.py restart
+```
+
+**Features:**
+- Runs server in background
+- Auto-detects Poetry or system Python environment
+- Manages PID file for process tracking
+- Graceful shutdown with fallback to force kill
+
+**Default Configuration:**
+- Port: 8001 (configurable via `.env.backend`)
+- Auto-reload: Enabled for development
+
+### Using Poetry Directly
+
+For foreground execution with full output:
+
+```bash
+cd backend
+poetry run uvicorn backend.api.api:app --host 0.0.0.0 --port 8001 --reload
+```
+
+### Using Docker Compose (Full Stack)
+
+To run the complete stack including nginx, webhooks, and MongoDB:
+
```

Comment:
Note that webhooks is really just a second thread of backend. It's the same image, started twice

Speaking of... Can you please file two issues:
 - There's an API end point api/precompute that I call with curl from a for loop in a bash shell in screen... This should also move to a separate container like webhooks
 - The problem with webhooks is that it will deploy EC2 instances and ssh into them synchronously, blocking everything while doing that. So we really need webhooks/ to just receive github events, put them in a job queue in MongoDB, and then have one or more separate workers that pick up work from that queue.
 - Once we have that, the precompute api end point can be removed and given to the workers to run as a scheduled task / forever.